### PR TITLE
Added explicit dependencies option to createEpicMiddleware

### PR DESCRIPTION
**Pull request by bali182**
Implements #163 

- [x] I have read the [Contributor Guide](https://github.com/redux-observable/redux-observable/blob/master/CONTRIBUTING.md)
- [x] My commit messages are in [conventional-changelog-standard](https://github.com/redux-observable/redux-observable/blob/master/CONTRIBUTING.md#sending-a-pull-request) format.

### CODE REVIEW

**Comment by bali182** [comment]
@jayphelps let me know if I should add more tests or change something!

**Comment by jayphelps** [comment]
Wow awesome!! Impressed with the test coverage actually including the discussed combineEpics spread support.

Everything looks great, we just gotta decide if defaulting to `null` is the right choice instead of not passing anything at all. Any arguments either way?

**Comment by bali182** [comment]
Of course, I wasn't sure about `null` either, it could be `undefined` as well. I didn't really want to do anything opinionated like an empty object or array, but I'm open to change it to any other default value.

**Comment by BerkeleyTrue** [comment]
@bali182 Nice work. I'd definitely think leaving it undefined is a better option.

@jayphelps should the function avoid passing in the third option if it undefined?

**Comment by jayphelps** [comment]
@BerkeleyTrue I believe so, that way `arguments.length` is not 3 (which it is even if you pass `undefined` literally.) Kinda a pain, but I think it's the "right thing to do".

I've amended the PR with those changes. Can someone give a final review before I merge?

**Comment by jayphelps** [comment]
Actually, I think we should be a stickler about not merging this until it includes documentation, even if brief. I think another section under Basics, something like "Injecting Dependencies Into Epics" but @bali182 might have better suggestions?

**Comment by bali182** [comment]
@jayphelps Sounds good as a title, roughly that's how I googled it and found the open issue :)

If you want I could help with the docs, have some free time this afternoon.

**Comment by jayphelps** [comment]
@bali182 please do! can just amend this PR

**Comment by bali182** [comment]
@jayphelps Done! Please do look at [createEpicMiddleware.md](https://github.com/redux-observable/redux-observable/pull/193/files#diff-a7715f80ded1874f5f0084ebddf948d3) with extreme care, because I added some description about the `options` argument, and I'm not 100% sure you planned it this way. Let me know I should change it anywhere!

**Comment by jayphelps** [comment]
@bali182 awesome. I made some changes, can you give me your honest thoughts? https://github.com/redux-observable/redux-observable/pull/193/commits/e83a27393be76264eaa08c0fd60abc59e641fe9d

File: docs/recipes/InjectingDependenciesIntoEpics.md [hunk 1]
```diff
@@ -1,55 +1,79 @@
...
+
+const fetchUserEpic = (action$, store) =>
+  action$.ofType('FETCH_USER')
+    .mergeMap(({ payload }) =>
+      ajax.getJSON(`/api/users/${payload}`)
+        .map(response => ({
+          type: 'FETCH_USER_FULFILLED',
+          payload: response
+        }))
+    );
```
Comment (jayphelps) [inline docs/recipes/InjectingDependenciesIntoEpics.md]: I mostly changed this to fetching a single user (simpler as an example to grok) and adjusted some things to be consistent with the other doc examples like mergeMap vs flatMap, isolating the observable chains by putting the map inside the mergeMap.

Comment (jayphelps) [inline docs/recipes/InjectingDependenciesIntoEpics.md]: I mostly changed this to fetching a single user (simpler as an example to grok) and adjusted some things to be consistent with the other doc examples like mergeMap vs flatMap, isolating the observable chains by putting the map inside the mergeMap.

**Comment by bali182** [comment]
@jayphelps Read the whole thing, thanks for correcting it (especially the `Observable => any` and `any => Observable`, this is exactly why I asked to take a look :)! The recipe looks much clearer, having an actual example of how things can be mocked helps a lot. Also it helps that the wording is coming from a native English speaker. I'd say it's good. Only thing I don't understand is that why `mergeMap` is used in all examples. Coming from java/scala `flatMap` is much more self explanatory for me but that's just personal preference.

**Comment by jayphelps** [comment]
@bali182 thanks for reading.

Some backstory: mergeMap is preferred because that's what RxJS v5 prefers. The documentation, links, filenames, etc are all mergeMap, only referencing that flatMap is an alias. The justification of that change in RxJS is obviously not redux-observable related, but was primary because we discussed the naming of flatMap + flatMapLatest with the community and by and large people (who were not already familiar with the term flatMap) preferred mergeMap and that it was quicker and easier to understand what it did. _That said_...that decision was made like almost 2 years ago and since then casually discussed the possibility of going back to flatMap but sticking with switchMap (instead of flatMapLatest). But I wouldn't bet on it changing.. For now we just go with the canonical name RxJS v5 uses.

**Comment by bali182** [comment]
@jayphelps thanks for the explanation! Didn't know about the `mergeMap` preference in rxjs

**Comment by BerkeleyTrue** [comment]
Noice!

**Comment by jayphelps** [comment]
Released in 0.14.0 https://github.com/redux-observable/redux-observable/blob/master/CHANGELOG.md#0140-2017-03-02

**Comment by bali182** [comment]
Great! :tada: :tada: :tada:

**Pull request merged**